### PR TITLE
Responsive in app surveys

### LINKED ISSUE #544: [FEATURE] Make in-app surveys mobile friendly + preview in survey editor
**Is your feature request related to a problem? Please describe.**
As a user, I want to be sure that the widget looks good on mobile phones as well.

**Describe the solution you'd like**
1. Make sure the widget is fully responsive on all devices and mobile browsers
2. Build a mobile preview toggle into the survey editor

**Full Responsiveness**
grafik
grafik

- Make sure all question types look good
- Test several device widths
- Maybe this helps (free & open-source): [Responsively App](https://responsively.app/)
- Make sure the breakpoint to Tablets is done well
- It should slide in up from the bottom of the phone and slide out down on close
- The Multi / Single Select options should have less paddings to make it fit smaller screens better

**Update Survey Editor with Preview**
grafik
grafik

- Build a toggle to switch between desktop and mobile view in the Survey Editor. [Maybe this helps.](https://ui.shadcn.com/docs/components/tabs)
- Make sure it works for both types of surveys (in-app and link survey)

### How we code at Formbricks 🤓

- Everything is type-safe
- All UI components are in the package `formbricks/ui`
- Run `pnpm dev` to find a demo app to test in-app surveys at `localhost:3002`
- We use **chatGPT** to help refactor code. Use our [Formbricks ✨ megaprompt ✨](https://github.com/formbricks/formbricks/blob/main/megaprompt.md) to create the right context before you write your prompt.

### PR DESCRIPTION
**Pull request by Dhruwang**
## What does this PR do?

Makes in-app surveys mobile friendly
Adds functionality to preview survey in mobile device

Fixes #544 

https://github.com/formbricks/formbricks/assets/67850763/6edba914-72fb-405e-bc63-0972a93ba51b

https://github.com/formbricks/formbricks/assets/67850763/456c6022-443a-436a-88e7-371ad9df02e1

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

Create an in-app survey and test on small devices
Go to preview survey page and toggle between mobile view and desktop view

## Checklist

- [x] Added a screen recording or screenshots to this PR
- [x] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/js/src/components/Modal.tsx [hunk 1]
```diff
@@ -76,9 +99,12 @@ export default function Modal({
         <div
           ref={modalRef}
           className={cn(
+            "fb-bottom-0 fb-inset-x-0",
             getPlacementStyle(placement),
             show ? "fb-opacity-100" : "fb-opacity-0",
-            "fb-h-fit fb-pointer-events-auto fb-absolute fb-w-full fb-max-w-sm fb-overflow-hidden fb-rounded-lg fb-bg-white fb-shadow-lg fb-ring-1 fb-ring-black fb-ring-opacity-5 fb-transition-all fb-duration-500 fb-ease-in-out sm:fb-m-4"
+            "fb-h-fit fb-pointer-events-auto fb-absolute fb-w-full sm:fb-max-w-sm fb-overflow-hidden fb-rounded-lg fb-bg-white fb-shadow-lg fb-ring-1 fb-ring-black fb-ring-opacity-5 fb-transition-all fb-duration-500 fb-ease-in-out sm:fb-m-4",
```
Comment (Dhruwang) [inline packages/js/src/components/Modal.tsx:105]: Here fb-bottom-0 and fb-inset-x-0 makes the modal sticks to bottom on small devices

File: packages/js/src/components/Modal.tsx [hunk 2]
```diff
@@ -76,9 +99,12 @@ export default function Modal({
...
+            isMobile && "fb-top-full fb-rounded-t-3xl",
```
Comment (Dhruwang) [inline packages/js/src/components/Modal.tsx:106]: When on small screen size, modal stays below the screen initially and slides up when show value becomes true (next line)

**Comment by jobenjada** [comment]
Hey @Dhruwang 

thanks a lot for this PR, really appreciate the full description and comments on the code - huge improvement! 🙏 ⚡ 

I've noticed a few things:

- The mobile preview always shows the in-app survey:
https://tella.video/johanness-video-2d8g

- The toggle is too big, especially on small screens. Broadly speaking, half the size should be sufficient:
grafik

- The phone looks a bit clunky. Pls increase the height a bit (gray lines) and decrease the border width a bit (blue line):
grafik

Thanks for picking this one up - looks great! 😊

**Comment by Dhruwang** [comment]
On it 😊

**Comment by Dhruwang** [comment]
Changes:
Made mobile mockup border thinner
Increased mockup's height
Reduced toggle's size
Removed duplicates CSS classes
Fixed : The mobile preview always shows the in-app survey

https://github.com/formbricks/formbricks/assets/67850763/e53e6659-f92b-47cb-8195-db28c4075289

File: apps/web/app/environments/[environmentId]/surveys/PreviewSurvey.tsx [hunk 3]
```diff
@@ -24,6 +26,68 @@ interface PreviewSurveyProps {
...
+        thankYouCard={thankYouCard}
+        gotoNextQuestion={gotoNextQuestion}
+      />
+      {showFormbricksSignature && <FormbricksSignature />}
+    </div>
+    <Progress progress={progress} brandColor={brandColor} />
+  </div>
+);
+
+const QuestionRenderer = ({ activeQuestionId, lastActiveQuestionId, questions, brandColor, thankYouCard, gotoNextQuestion }) => {
```
Comment (Dhruwang) [inline apps/web/app/environments/[environmentId]/surveys/PreviewSurvey.tsx:60]: Extracted question rendering logic into a separate component as it was being used multiple times

**Comment by jobenjada** [comment]
Hey! 

We're getting there 🙏  I found a few more small tweaks / bugs:

1. Survey does not jump back to top in preview when two questions with scroll follow behind each other:
https://tella.video/johanness-video-3qzv

2. Reduce height of nodge a bit, there is not enough space between the question and the nodge
grafik

3. Max height of Link Survey container with too high to fully display toggle:
grafik

The rest looks good! 🚀

**Comment by Dhruwang** [comment]
On it 😊

**Comment by Dhruwang** [comment]
Resolved above mentioned issues

https://github.com/formbricks/formbricks/assets/67850763/aef6e463-cef0-4720-8e12-c9591fb222b5


https://github.com/formbricks/formbricks/assets/67850763/69616c9b-4fed-453c-b482-6d5a759be204

**Comment by jobenjada** [comment]
Hey @Dhruwang  !

Looks great, works smoothly 🚀  Can you please resolve the two merge conflicts?

@mattinannt from UI/UX ready to merge :)

**Comment by Dhruwang** [comment]
Merge conflicts resolved
Also fixes a minor issue of back button disappearing from preview once survey is finished

https://github.com/formbricks/formbricks/assets/67850763/2555f0cc-d496-4298-a097-9649b94803c1

**Comment by mattinannt** [comment]
@Dhruwang thanks for the PR 😊 It looks great in your video but when testing on my computer in the demo app, the popup is showing bottom left instead of bottom right and on mobile the popup isn't displayed at all (no errors in console, just not showing (css)). Can you please take a look? 😊

https://github.com/formbricks/formbricks/assets/675065/023a7cf7-8456-4501-8164-5f1ab7dda3b8

**Comment by Dhruwang** [comment]
Have resolved the issue 😊 @mattinannt

**Review by mattinannt — APPROVED** [review]
@Dhruwang thanks a lot for making the changes 😊 The new preview looks really great and now also the web-app survey is working correctly 💪

I will merge the changes into main now 😊

**Pull request merged**